swim/handlers_test.go

```diff
@@ -92,6 +92,13 @@ func (s *HandlerTestSuite) TestGossipStopHandler() {
 	s.True(s.testNode.node.gossip.Stopped())
 }
 
+func (s *HandlerTestSuite) TestPartitionHealerHandler() {
+	ExecuteThenWaitFor(func() {
+		_, err := s.testNode.node.partitionHealerHandler(s.ctx, &emptyArg{})
+		s.NoError(err, "calling handler should not result in error")
+	}, s.testNode.node, DiscoHealEvent{})
```

Comment:
Disagree with the previous comment. If the test hangs, that is really bad and annoying. Imagine on Travis: the CI result would take 10 minutes to come in and blocks everything.

Considering how easy it is to add a timeout (2-3 lines of code?) I think we should add a timeout that fails the test when the timeout hits.